web/src/compose_actions.ts

```diff
@@ -25,28 +27,30 @@ import * as spectators from "./spectators";
 import {realm} from "./state_data";
 import * as stream_data from "./stream_data";
 
-const compose_clear_box_hooks = [];
-const compose_cancel_hooks = [];
+type Hook = () => void;
```

Comment:
This is a type we'll export, so should have a name that feels readable when imported in other modules. `ComposeHook` would be appropriate, or maybe even having two separate types, `ComposeCancelHook` and `ComposeClearBoxHook`, with equal values, could be OK.

(Generally, our project is small enough that picking type names that will be globally unique in the codebase is a good idea).

Reply:
That makes sense, thanks. I went with `ComposeHook` for now.